docs/docs/querying.md

```diff
@@ -299,3 +299,150 @@ something.findOne({
   order: sequelize.literal('convert(user_name using gbk)')
 })
 ```
+
+## CTEs and Recursive Queries
+
+CTEs (Common Table Expressions) are special expressions that preceded a query. They create a named expression that can be referenced later in a query. Some dialects allow a CTE to perform a recursive query, a query that repeats itself as long as it is able to. There may be many CTEs with a single query, and each CTE can build on the results of the one that came before.
+
+
+`cte` takes an array of objects that each describe a common table expression or CTE. CTEs run before the main query and can be made recursive.
+
+```js
+something.findAll({
+    cte: [{
+        name: 'a',
+        intial: {
+            where: {
+                name: 'Jane'
+            }
+        },
+        recursive: {
+            next: 'friend'
+        },
+        limit: 20
+    }]
+    cteSelect: 'a'
+}
+```
+
+### Creating a CTE
+
+CTEs are usually based on the model being queried. This happens automatically, but can be controlled through `useModel` and `model`.
+
+```js
+cte: [{
+    name: 'b',
+    model: User,
+    useModel: true
+}]
```

Comment:
As of this moment the where would need to be placed in the `initial` object, but I am now looking into providing some shortcuts to avoid the idea of `initial` and `recursive` for people who don't want to use CTEs in a recursive way.